docs/examples.md

```diff
@@ -0,0 +1,273 @@
+# Examples
+
+The MCP PHP SDK includes comprehensive examples demonstrating different patterns and use cases. Each example showcases specific features and can be run independently to understand how the SDK works.
+
+## Table of Contents
+
+- [Getting Started](#getting-started)
+- [Running Examples](#running-examples)
+- [STDIO Examples](#stdio-examples)
+- [HTTP Examples](#http-examples)
+- [Advanced Patterns](#advanced-patterns)
+- [Testing and Debugging](#testing-and-debugging)
+
+## Getting Started
+
+All examples are located in the `examples/` directory and use the SDK dependencies from the root project. Most examples can be run directly without additional setup.
+
+### Prerequisites
+
+```bash
+# Install dependencies (from project root)
+composer install
+
+# Optional: Install HTTP-specific dependencies for web examples
+composer install --dev
+```
+
+## Running Examples
+
+### STDIO Examples
+
+STDIO examples use standard input/output for communication:
+
+```bash
+# Interactive testing with MCP Inspector
+npx @modelcontextprotocol/inspector php examples/01-discovery-stdio-calculator/server.php
+
+# Run with debugging enabled
+npx @modelcontextprotocol/inspector -e DEBUG=1 -e FILE_LOG=1 php examples/01-discovery-stdio-calculator/server.php
+
+# Or configure the script path in your MCP client
+# Path: php examples/01-discovery-stdio-calculator/server.php
+```
+
+### HTTP Examples
+
+HTTP examples run as web servers:
+
+```bash
+# Start the server
+php -S localhost:8000 examples/02-discovery-http-userprofile/server.php
+
+# Test with MCP Inspector
+npx @modelcontextprotocol/inspector http://localhost:8000
+
+# Test with curl
+curl -X POST http://localhost:8000 \
+  -H "Content-Type: application/json" \
+  -H "Accept: application/json, text/event-stream" \
+  -d '{"jsonrpc":"2.0","id":1,"method":"initialize","params":{"protocolVersion":"2024-11-05","clientInfo":{"name":"test","version":"1.0.0"},"capabilities":{}}}'
+```
+
+## STDIO Examples
+
+### 01. Discovery STDIO Calculator
+
+**File**: `examples/01-discovery-stdio-calculator/`
+
+**What it demonstrates:**
+- Attribute-based discovery using `#[McpTool]` and `#[McpResource]`
+- Basic arithmetic operations
+- Configuration management through resources
+- State management between tool calls
+
+**Key Features:**
+```php
+#[McpTool(name: 'calculate')]
+public function calculate(float $a, float $b, string $operation): float|string
+
+#[McpResource(
+    uri: 'config://calculator/settings',
+    name: 'calculator_config',
+    mimeType: 'application/json'
+)]
+public function getConfiguration(): array
+```
+
+**Usage:**
+```bash
+# Interactive testing
+npx @modelcontextprotocol/inspector php examples/01-discovery-stdio-calculator/server.php
+
+# Or configure in MCP client: php examples/01-discovery-stdio-calculator/server.php
+```
+
+### 03. Manual Registration STDIO
```

Comment:
somehow i struggle with the term "manual" registration - i can see where it's coming from, but it's basically just a set of adder methods. what do you think about using "explicit" instead?